File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/witness/WitnessPagerAdapter.java
package com.github.dedis.popstellar.ui.lao.witness;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class WitnessPagerAdapter extends FragmentStateAdapter {

  public WitnessPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
    super(fragmentActivity);
  }

  @NonNull
  @Override
  public Fragment createFragment(int position) {
    if (position == 0) {
      return new WitnessesFragment();
    } else {
      return new WitnessMessageFragment();
    }
  }

  @Override
  public int getItemCount() {
    return 2; // We have 2 tabs
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/witness/WitnessPagerAdapter.kt
package com.github.dedis.popstellar.ui.lao.witness

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class WitnessPagerAdapter(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
  override fun createFragment(position: Int): Fragment {
    return if (position == 0) {
      WitnessMessageFragment()
    } else {
      WitnessesFragment()
    }
  }

  override fun getItemCount(): Int {
    return 2 // We have 2 tabs
  }
}
